components/net_traits/lib.rs

```diff
@@ -491,14 +491,23 @@ impl Metadata {
 }
 
 /// The creator of a given cookie
-#[derive(PartialEq, Copy, Clone, Deserialize, Serialize)]
+#[derive(PartialEq, Clone, Deserialize, Serialize)]
 pub enum CookieSource {
     /// An HTTP API
-    HTTP,
+    HTTP(ServoUrl),
     /// A non-HTTP API
     NonHTTP,
```

Comment:
I think we're going to have to add a ServoUrl to this case to, or we'll break allowing pages to set secure cookies via document.cookie when loaded via a secure scheme.